src/annotator/plugin/test/document-test.coffee

```diff
@@ -0,0 +1,171 @@
+      assert.equal(metadata.link[7].type, "application/pdf")
+      assert.equal(metadata.link[8].href, "doi:10.1175/JCLI-D-11-00015.1")
+
+    it 'should ignore atom and RSS feeds and alternate languages', ->
+      assert.equal(metadata.link.length, 9)
+
+    it 'should have highwire metadata', ->
+      assert.ok(metadata.highwire)
+      assert.deepEqual(metadata.highwire.pdf_url, ['foo.pdf'])
+      assert.deepEqual(metadata.highwire.doi, ['10.1175/JCLI-D-11-00015.1'])
+      assert.deepEqual(metadata.highwire.title, ['Foo'])
+
+    it 'should have dublincore metadata', ->
+      assert.ok(metadata.dc)
+      assert.deepEqual(metadata.dc.identifier, ["doi:10.1175/JCLI-D-11-00015.1", "isbn:123456789"])
+      assert.deepEqual(metadata.dc.type, ["Article"])
+
+    it 'should have facebook metadata', ->
+      assert.ok(metadata.facebook)
+      assert.deepEqual(metadata.facebook.url, ["http://example.com"])
+
+    it 'should have eprints metadata', ->
+      assert.ok(metadata.eprints)
+      assert.deepEqual(metadata.eprints.title, ['Computer Lib / Dream Machines'])
+
+    it 'should have prism metadata', ->
+      assert.ok(metadata.prism)
+      assert.deepEqual(metadata.prism.title, ['Literary Machines'])
+
+      it 'should have twitter card metadata', ->
+        assert.ok(metadata.twitter)
+        assert.deepEqual(metadata.twitter.site, ['@okfn'])
+
+    it 'should have unique uris', ->
+      uris = testDocument.uris()
+      assert.equal(uris.length, 7)
+
+    it 'uri() returns the canonical uri', ->
+      uri = testDocument.uri()
+      assert.equal(uri, metadata.link[5].href)
+
+    it 'should have a favicon', ->
+      assert.equal(
+        metadata.favicon
+        'http://example.com/images/icon.ico'
+      )
+
+  describe '#_absoluteUrl', ->
+
+    it 'should add the protocol when the url starts with two slashes', ->
+      result = testDocument._absoluteUrl('//example.com/')
+      expected = "#{document.location.protocol}//example.com/"
+      assert.equal(result, expected)
+
+    it 'should add a trailing slash when given an empty path', ->
+      result = testDocument._absoluteUrl('http://example.com')
+      assert.equal(result, 'http://example.com/')
+
+    it 'should make a relative path into an absolute url', ->
+      result = testDocument._absoluteUrl('path')
+      expected = (
+        document.location.protocol + '//' +
+          document.location.host +
+          document.location.pathname.replace(/[^\/]+$/, '') +
+          'path'
+      )
+      assert.equal(result, expected)
+
+    it 'should make an absolute path into an absolute url', ->
+      result = testDocument._absoluteUrl('/path')
+      expected = (
+        document.location.protocol + '//' +
+          document.location.host +
+          '/path'
+      )
+      assert.equal(result, expected)
+
+  describe '#_getDocumentHref', ->
+
+    it 'should use the baseURI for the current testing setup', ->
```

Comment:
I suggest this description could be clearer about what is being tested, eg. "should use the baseURI when the protocol is different from the document location".

In this case it wasn't immediately obvious to me that the reason this returns the base URI here is because the test simulates having an `https://` base URI but the document is being served by Karma with an `http://` URL.